python/pyarrow/compat.py

```diff
@@ -102,7 +102,7 @@ def tobytes(o):
             return o
 
     def frombytes(o):
-        return o
+        return str(o)
```

Comment:
This seems like a workaround for a different issue -- the idea here is that `o` is supposed to be bytes already, so on Py2 this is a no-op

Reply:
The reason I did this was because it's being used like `str` in other places, which I guess technically isn't the correct way to use it.

Reply:
No, this function is only to be used for str-izing something we know to be bytes. We can make some other kind of string coercion compat function

Reply:
Yep, I'm going to start from scratch on the problem because the original PR doesn't solve the problem correctly.